backend/api.py

```
@@ -1930,12 +1964,25 @@ async def validate_path(body: dict, request: Request, _=Depends(verify_local_req
     if not path:
         return {"valid": False, "error": "Path is required"}
 
-    # Normalize to prevent path traversal tricks
-    import pathlib
+    # Normalize with pure string operations (no filesystem call sees the raw
+    # request value), then require the result to sit under an allowed root.
     try:
-        normalized = str(pathlib.Path(path).resolve())
-    except (ValueError, OSError):
+        normalized = os.path.abspath(os.path.normpath(path))
```

Ruff (0.15.21)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)